client/src/components/Workflow/WorkflowList.vue

```diff
@@ -213,8 +207,22 @@ export default {
             };
             this.services
                 .updateWorkflow(id, data)
-                .then(() => {
-                    window.location = `${getAppRoot()}workflows/list`;
+                .then(({ id, name }) => {
+                    if (checked) {
+                        const indexToRemove = getGalaxyInstance().config.stored_workflow_menu_entries.findIndex(
+                            (workflow) => workflow.id === id
+                        );
+                        getGalaxyInstance().config.stored_workflow_menu_entries.splice(indexToRemove, 1);
+                    } else {
+                        getGalaxyInstance().config.stored_workflow_menu_entries.push({ id: id, name: name });
+                    }
```

Comment:
I can't wait for the day that we have an outer parent component for the whole app that could receive config events emitted here. Or maybe the config provider could listen to such events ?

Reply:
This is what vuex store is for, which should be a singleton across components (whether in the same tree or not).

The store is sync'd do the global galaxy object as a convenience for other cruft that still uses it, but there should not be a need to access it like this right now.  Just add a workflow update action and dispatch it here.